tests/fips/test_fips.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import re
+
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.sanity_check(skip_sanity=True),
+    pytest.mark.disable_loganalyzer,  # disable automatic loganalyzer
+    pytest.mark.topology('any')
+]
+
+
+def test_fips(duthosts, enum_rand_one_per_hwsku_hostname):
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+
+    # check if the fips feature enabled on image level:
+    fips_status = duthost.shell("sudo sonic-installer get-fips")["stdout"]
+    if "FIPS is enabled" not in fips_status:
+        logger.info("Ignore test becauser FIPS no enabled")
+
+    # check kernel parameter
+    kernel_cmdline = duthost.shell("cat /proc/cmdline")["stdout"]
+    logger.warning("Kernel cmdline: {}".format(kernel_cmdline))
+    pytest_assert("sonic_fips=1" in kernel_cmdline,
+                  "Not found sonic_fips enable flag: {}".format(kernel_cmdline))
+
+    # check FIPS package and symcrypt package installed
+    fips_packages = duthost.shell("sudo apt list --installed | grep fips")["stdout"]
+    logger.warning("Fips package: {}".format(fips_packages))
+    pytest_assert(re.search("libk5crypto3.*fips", fips_packages),
+                  "libk5crypto3 missing: {}".format(fips_packages))
+    pytest_assert(re.search("libssl-dev.*fips", fips_packages),
+                  "libssl missing: {}".format(fips_packages))
+    pytest_assert(re.search("libssl3.*fips", fips_packages),
+                  "libssl3 missing: {}".format(fips_packages))
+    pytest_assert(re.search("openssh-client.*fips", fips_packages),
+                  "openssh client missing: {}".format(fips_packages))
+    pytest_assert(re.search("openssh-server.*fips", fips_packages),
+                  "openssh server missing: {}".format(fips_packages))
+    pytest_assert(re.search("openssh-sftp-server.*fips", fips_packages),
+                  "openssh sftp server missing: {}".format(fips_packages))
+    pytest_assert(re.search("openssl.*fips", fips_packages),
+                  "openssl missing: {}".format(fips_packages))
+    pytest_assert(re.search("ssh.*fips", fips_packages),
+                  "ssh missing: {}".format(fips_packages))
+    pytest_assert(re.search("libpython3.*fips", fips_packages),
+                  "libpython3 missing: {}".format(fips_packages))
+
+    symcrypt_packages = duthost.shell("sudo apt list --installed | grep symcrypt")["stdout"]
+    logger.warning("symcrypt package: {}".format(symcrypt_packages))
+    pytest_assert(re.search("symcrypt-openssl.*", symcrypt_packages),
+                  "symcrypt-openssl missing: {}".format(symcrypt_packages))
+
+    # check symcrypt engine loaded
+    openssl_engines = duthost.shell("openssl engine -vv | grep -i symcrypt")["stdout"]
```

Comment:
>openssl engine -vv [](http://example.com/codeflow?start=37&length=18)

@xincunli-sonic Is it working on every hwsku?

Reply:
This is an openssl command, It works on those 5 HwSku.

Reply:
confirmed, working.